build: Add unit test target generation to `cpp_library` CMake util; Add tasks for building and running unit tests. (#26)
y-scope/ystdlib-cpp: README.md

```

## Building
To build all targets:
```shell
task build:all
```

> To build the global unit test target:

Reviewer: ```suggestion
To build a unified executable containing all unit tests:
```

```shell
task build:unit-test
```

> To build unit tests for a specific library:
> ```shell
> task build:unit-test -- <lib_name>

Reviewer: ```suggestion
To build individual unit test executables for a subset of libraries:
```shell
task build:unit-test -- <lib_name0> <lib_name1> ...
```

```

## Testing
To build and run all unit tests:
```shell
task test
```

> To build and run unit tests for a specific library:
> ```shell
> task test -- <lib_name>

Reviewer: ```suggestion
To build and run unit tests for a subset of libraries:
```shell
task test -- <lib_name0> <lib_name1> ...
```

```

## Building
To build all targets:
```shell
task build:all
```

To build an executable containing all unit tests:
```shell
task build:unit-test-all
```

To build an executable containing a single library's unit tests:
```shell
task build:unit-test-<lib_name>
```

## Testing

> To build and run unit tests for a specific library:
> ```shell
> task test-<lib_name>

Reviewer: ```suggestion
To build and run all unit tests:
```shell
task test-all
```

```

> #### Special Targets
> To build and run all unit tests:
> ```shell
> task test-all

Reviewer: ```suggestion
To build and run unit tests for a specific library:
```shell
task test-<lib_name>
```
